docs/usage/keycloak.md

```diff
@@ -0,0 +1,37 @@
+# Usage
+
+The following example shows the most minimal valid manifest to create a new Argo CD cluster with keycloak as Single sign-on provider.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ArgoCD
+metadata:
+  name: example-argocd
+  labels:
+    example: basic
+spec:
+  sso:
+    provider: keycloak
+```
+
+## Create
+
+Create a new Argo CD cluster in the `argocd` namespace using the provided basic example.
+
+```bash
+kubectl create -n argocd -f examples/argocd-basic.yaml
+```
+
+The above configuration creates a keycloak instance and its relevant resources along with the Argo CD resources. Users can login into the keycloak console using the below commands.
+
+Get the Keycloak Route URL for Login.
+
```

Comment:
Route is openshift specific.   I wonder if we create an ingress for kubernetes clusters?

Reply:
updated the note and updated the doc for openshift only. we will create a different section for k8s as we use ingress in that case and also usage steps might be slightly different.